subsys/bluetooth/host/gatt.c

```diff
@@ -99,11 +99,11 @@ static ssize_t write_name(struct bt_conn *conn, const struct bt_gatt_attr *attr,
 {
 	char value[CONFIG_BT_DEVICE_NAME_MAX] = {};
 
-	if (offset) {
+	if (offset > sizeof(value)) {
 		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
 	}
 
-	if (len >= sizeof(value)) {
+	if (offset + len > sizeof(value)) {
```

Comment:
Shouldn't these comparisons be `>=` so that you guarantee that there's a null byte at the end?

Reply:
The reason being that `value` gets later passed to `bt_set_name()` as a normal null-terminated string. We'd open ourself for potential buffer overflow exploits if we don't ensure that the result is null-terminated.

Reply:
Yes, they should :) I pushed the change